src/corona_hakab_model/infection.py

```diff
@@ -52,4 +59,18 @@ def _perform_infection(self) -> List[Agent]:
             agent = self.manager.agents[index]
             new_infected.append(agent)
 
-        return new_infected
+        return new_infected, infection_methods
+
+    def _get_infection_method(self, agent_index, non_zero_columns):
```

Comment:
this method ignores the medical states of the possible infectors.
should use the contagiousness vector either in selecting possible infectors, or in the probability calculation

Reply:
fixed